components/quick_input/clearable_tooltip.tsx

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Tooltip} from 'react-bootstrap';
+
+import OverlayTrigger from 'components/overlay_trigger';
+import Constants from 'utils/constants.jsx';
+
+interface IClearableTooltipProps{
+    clearableTooltipText?: string | JSX.Element;
+    clearable?: boolean;
+    onClear?: (e: React.MouseEvent) => void;
+    value: string;
+}
+
+const ClearableTooltip = (props: IClearableTooltipProps) => {
+    const clearableTooltipText = React.useMemo(() => {
```

Comment:
Do we need this hook for the tooltip component?

Reply:
I suggest splitting it into separate components. The quick input component files is confusing enough =))

Reply:
Sounds good 👍